File: hw3/ArrayQueue.java
import java.util.ArrayList;

public class ArrayQueue implements Queue {
    ArrayList<Object> arrayList = new ArrayList<Object>();

    void add(Object o) {
        arrayList.add(o);
    }

    Object remove() {
        return arrayList.remove(0);
    }

    boolean isEmpty() {
        return arrayList.isEmpty();
    }
}

File: hw4/ArrayQueue.cpp
// Author: Logan Tillman

#include <vector>
#include "Queue.h"

using namespace std;

void ArrayQueue::add(int value) {
	vec.push_back(value);
}
	
int ArrayQueue::remove() {
	int rv = vec.at(0);
	vec.erase(vec.begin());
	return rv;
}

bool ArrayQueue::isEmpty() {
	if (vec.size() == 0)
		return true;
	else
		return false;
}

